build/circle-test.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+set -e
+
+outdir="${TMPDIR}"
+if [ "${CIRCLE_ARTIFACTS}" != "" ]; then
+    outdir="${CIRCLE_ARTIFACTS}"
+fi
+
+builder=$(dirname $0)/builder.sh
+
+# 1. Run "make check" to verify coding guidelines.
+echo "make check"
+time ${builder} make GITHOOKS= check | tee "${outdir}/check.log"; test ${PIPESTATUS[0]} -eq 0
+
+# 2. Verify that "go generate" was run.
+echo "verifying generated files"
+time ${builder} /bin/bash -c "(go generate ./... && git ls-files --modified --deleted --others --exclude-standard | diff /dev/null -) || (git add -A && git diff -u HEAD && false)" | tee "${outdir}/generate.log"; test ${PIPESTATUS[0]} -eq 0
+
+# 3. Run "make testrace".
+match='^panic|^[Gg]oroutine \d+|(read|write) by.*goroutine|DATA RACE'
+echo "make testrace"
+time ${builder} make GITHOOKS= testrace \
```

Comment:
With pipelines, only the last command's status counts unless you also `set -o pipefail`. Tamir did this for a bunch of our scripts recently. But we don't want to fail as soon as `make testrace` fails, because we still want the xml postprocessing to occur. We need to capture the exit status without failing the whole script, or split things up into separate circle.yml statements (this is how we currently run the xml step regardless of the status of `make`)